openedx/core/djangoapps/content/course_overviews/signals.py

```diff
@@ -111,19 +161,62 @@ def _check_for_pacing_changes(previous_course_overview, updated_course_overview)
+    the system are aware of the change. The `credentials` app listens for this signal in order to keep our certificate
+    visibility settings in the Credentials IDA up to date.
+
+    Args:
+        previous_course_overview (CourseOverview): the current course overview instance for a particular course run
+        updated_course_overview (CourseOverview): an updated course overview instance, reflecting the current state of
+            data from the modulestore/Mongo
+
+    Returns:
+        None
+    """
+    def _send_course_cert_date_change_signal():
+        """
+        A callback used to fire the COURSE_CERT_DATE_CHANGE Django signal *after* the ORM has successfully commited the
+        update.
+        """
+        COURSE_CERT_DATE_CHANGE.send_robust(sender=None, course_key=str(updated_course_overview.id))
+
+    course_run_id = str(updated_course_overview.id)
+    prev_available_date = previous_course_overview.certificate_available_date
+    prev_display_behavior = previous_course_overview.certificates_display_behavior
+    prev_end_date = previous_course_overview.end  # `end_date` is a deprecated field, use `end` instead
+    updated_available_date = updated_course_overview.certificate_available_date
+    updated_display_behavior = updated_course_overview.certificates_display_behavior
+    updated_end_date = updated_course_overview.end  # `end_date` is a deprecated field, use `end` instead
+    send_signal = False
+
+    if prev_available_date != updated_available_date:
         LOG.info(
-            f"Certificate availability date for {str(updated_course_overview.id)} has changed from " +
-            f"{previous_course_overview.certificate_available_date} to " +
-            f"{updated_course_overview.certificate_available_date}. Sending COURSE_CERT_DATE_CHANGE signal."
+            f"The certificate available date for {course_run_id} has changed from {prev_available_date} to "
+            f"{updated_available_date}"
         )
+        send_signal = True
 
-        def _send_course_cert_date_change_signal():
-            COURSE_CERT_DATE_CHANGE.send_robust(
-                sender=None,
-                course_key=updated_course_overview.id,
-            )
+    if prev_display_behavior != updated_display_behavior:
+        LOG.info(
+            f"The certificates display behavior for {course_run_id} has changed from {prev_display_behavior} to "
+            f"{updated_display_behavior}"
+        )
+        send_signal = True
+
+    # edge case -- if a course run with a cert display behavior of "End date of course" has changed its end date, we
+    # should fire our signal to ensure visibility of certificates managed by the Credentials IDA are corrected too
+    if (
+        (prev_display_behavior == CertificatesDisplayBehaviors.END and
+         updated_display_behavior == CertificatesDisplayBehaviors.END) and
```

Comment:
you still don't need both of these conditionals. You are already sending a signal if they are different  on lines 202-207.  you only need to know if the updated display behavior is END and  the end date has changed. But this is nonblocking; it's unnecessary but fine.

Reply:
I'll make this update before merging too, thanks for your patience here. I thought I took care of this yesterday with other changes and got distracted. This is sound and appreciate the time and effort you spent reviewing the code.